### PR TITLE
feat(Winston-logger): add multiple escalation paths depending on message context

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

This PR adds the ability for Winston logs to be directed to different "paths" enabling more granular control over where messages are sent.

**Summary**

- Add two main paths: `notificationPath: "risk-management"` & `notificationPath: "infrastructure-error"`
- Refactor how configs are fed into these transports. NOTE this is a breaking change and updates to configs must be rolled out at the same time!

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [X]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/financial-templates-lib/src/logger/PagerDutyTransport.js [hunk 1]
```diff
@@ -10,22 +10,29 @@ const pdClient = require("node-pagerduty");
...
   }
 
   async log(info, callback) {
     try {
       // If the message has markdown then add it and the bot-identifer field. Else put the whole info object as a string
       const logMessage = info.mrkdwn ? info.mrkdwn + `\n${info["bot-identifier"]}` : JSON.stringify(info);
 
+      // If the log contains a notification path then use a custom PagerDuty service. This lets the transport route to
+      // diffrent pagerduty escilation paths depending on the context of the log.
+      const serviceId = this.customServices[info.notificationPath] ?? this.defaultServiceId;
```
Comment (nicholaspai) [inline packages/financial-templates-lib/src/logger/PagerDutyTransport.js:28]: I've never seen this ?? syntax before, that's useful to know.

File: packages/financial-templates-lib/src/logger/PagerDutyTransport.js [hunk 2]
```diff
@@ -10,22 +10,29 @@ const pdClient = require("node-pagerduty");
...
+    this.customServices = pagerDutyOptions.customServices || {};
   }
 
   async log(info, callback) {
     try {
       // If the message has markdown then add it and the bot-identifer field. Else put the whole info object as a string
       const logMessage = info.mrkdwn ? info.mrkdwn + `\n${info["bot-identifier"]}` : JSON.stringify(info);
 
+      // If the log contains a notification path then use a custom PagerDuty service. This lets the transport route to
+      // diffrent pagerduty escilation paths depending on the context of the log.
```
Comment (nicholaspai) [inline packages/financial-templates-lib/src/logger/PagerDutyTransport.js:27]:
```suggestion
      // different pagerduty escilation paths depending on the context of the log.
```

;)

**Review by nicholaspai — APPROVED** [review]
Pleasantly surprised how succinct this feature is in terms of lines of code. LGTM after you resolve my nits

File: packages/financial-templates-lib/src/logger/SlackTransport.js [hunk 3]
```diff
@@ -169,26 +169,31 @@ class SlackHook extends Transport {
...
     this.mrkdwn = opts.mrkdwn || false;
 
     this.axiosInstance = axios.create({ proxy: opts.proxy || undefined });
   }
 
   async log(info, callback) {
+    // If the log contains a notification path then use a custom slack webhook service. This lets the transport route to
+    // diffrent slack channels depending on the context of the log.
+    const webhookUrl = this.escilationPathWebhookUrls[info.notificationPath] ?? this.defaultWebHookUrl;
+    // delete info.notificationPath; // Dont log the notification path.
```
Comment (mrice32) [inline packages/financial-templates-lib/src/logger/SlackTransport.js:184]: nit: is this supposed to be commented out?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/common/src/TransactionUtils.js [hunk 4]
```diff
@@ -39,7 +39,10 @@ const runTransaction = async ({ web3, transaction, transactionConfig, availableA
   if (await accountHasPendingTransactions(web3, transactionConfig.from))
     transactionConfig.nonce = await getPendingTransactionCount(web3, transactionConfig.from);
   // Else, there is no pending transaction and we use the current account transaction count as the nonce.
-  else transactionConfig.nonce = await web3.eth.getTransactionCount(transactionConfig.from);
+  // this method does not play niceley in tests. leave the nounce null to auto fill.
+  else if (argv.network != "test") transactionConfig.nonce = await web3.eth.getTransactionCount(transactionConfig.from);
+
+  console.log("args", argv);
```
Comment (daywiss) [inline packages/common/src/TransactionUtils.js:45]: not sure if this is intended

**Review by daywiss — APPROVED** [review]
lg

File: packages/common/src/TransactionUtils.js (hunk 4 above)
Comment (chrismaree) [reply packages/common/src/TransactionUtils.js]: was not. thanks for catching. have removed

**Pull request merged**